packages/ag-charts-enterprise/src/features/error-bar/errorBarNode.ts

```diff
@@ -44,14 +65,64 @@ export class ErrorBarNode extends _Scene.Group {
         return Math.min(desiredLength, lengthMax);
     }
 
-    updateStyle(style: AgErrorBarThemeableOptions) {
+    private getFormatterParams(
+        formatters: { formatter?: ErrorBarFormatter; cap: { formatter?: ErrorBarCapFormatter } } & AgErrorBarDataOptions
+    ): AgErrorBarFormatterParams | undefined {
+        const { datum } = this;
+        if (datum === undefined || (formatters.formatter === undefined && formatters.cap.formatter === undefined)) {
+            return undefined;
+        }
+        const { xLowerKey, xLowerName, xUpperKey, xUpperName, yLowerKey, yLowerName, yUpperKey, yUpperName } =
+            formatters;
+        return {
+            ...datum,
+            xLowerKey,
+            xLowerName,
+            xUpperKey,
+            xUpperName,
+            yLowerKey,
+            yLowerName,
+            yUpperKey,
+            yUpperName,
+        };
+    }
+
+    private applyStyling(target: AgErrorBarStylingOptions, source?: AgErrorBarStylingOptions) {
+        // Style can be any object, including user data (e.g. formatter
+        // result). So filter out anything that isn't styling options:
+        _ModuleSupport.partialAssign(
+            ['visible', 'stroke', 'strokeWidth', 'strokeOpacity', 'lineDash', 'lineDashOffset'],
+            target,
+            source
+        );
+    }
+
+    updateStyle(
+        style: AgErrorBarThemeableOptions,
+        formatters: { formatter?: ErrorBarFormatter; cap: { formatter?: ErrorBarCapFormatter } } & AgErrorBarDataOptions
+    ) {
         const { whiskerPath, capsPath } = this;
         const { cap, ...whiskerStyle } = style;
         const { length, lengthRatio, ...capsStyle } = cap ?? {};
-        Object.assign(whiskerPath, whiskerStyle);
-        Object.assign(capsPath, capsStyle);
+        this.applyStyling(whiskerPath, whiskerStyle);
+        this.applyStyling(capsPath, capsStyle);
         whiskerPath.markDirty(whiskerPath, _Scene.RedrawType.MINOR);
         capsPath.markDirty(capsPath, _Scene.RedrawType.MINOR);
+
+        const params = this.getFormatterParams(formatters);
+        if (params !== undefined) {
+            if (formatters.formatter !== undefined) {
+                const result = formatters.formatter(params);
+                this.applyStyling(whiskerPath, result);
+                this.applyStyling(capsPath, result);
+                this.applyStyling(capsPath, result?.cap);
+            }
+
+            if (formatters.cap.formatter !== undefined) {
+                const result = formatters.cap.formatter(params);
+                this.applyStyling(capsPath, result);
```

Comment:
`capsPath` is having settings applied multiple times - it would be better to calculate the styles and apply once, as the current approach has the possibility of triggering false-positive change-detection due to flip-flopping on style changes between the different inputs.

Reply:
That's a good idea. I've repurposed `mergeDefaults` from the box-plot series to avoid calling `applyStyling` more than once for each node.